processor/awscontainerinsightprocessor/README.md

```diff
@@ -0,0 +1,169 @@
+# AWS Container Insights Processor
+
+## Overview
+AWS Container Insights Processor (`awscontainerinsightprocessor`) is an AWS specific processor that supports [CloudWatch Container Insights]((https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights.html)). It decorates the metrics emitted by `awscontainerinsightreceiver` to support the same CloudWatch Container Insights experience.
+
+## Design of AWS Container Insights Processor for EKS
+
+* pod store:
+  * access Kubelet to list all Pod objects running on the same node, and cache all the Pod objects in memory
+  * find corresponding pod objects from the cache to decorate pod and container metrics
+  * information used to decorate pod and container metrics:
+    1. Pod Owner Reference
+    2. Container Resource Limit
+    3. Container Resource Request
+    4. Container ID and Status
+    5. Pod Status
+    6. Pod Labels
+* service store: 
+  * add the corresponding attribute `Service` to the metrics for the relevant pods
+  * call "List & Watch" for "Endpoint" objects from Kubernetes API server. In this mode, when there is any change to “Endpoint”, agent will get the notification of the change. The benefit is that the agent could get Endpoint information update in time. If the "Endpoint" resources just occasionally slightly changes, the impact to API server is supposed to be minimal.
+  * If customer is not interested in aggregating metrics per ServiceName, we allow them to disable the service decoration through the configuration.
+
+
+## Available Metrics and Resource Attributes
+
+The AWS Container Insights Processor decorates metrics differently depending on their types.
```

Comment:
These tables are somewhat hard to follow.  It seems like the third column, "Added Resource Attributes" is not correlated to the data in the rows of the first two columns.  If that is the case perhaps there should be two tables per section, or a table and a list.

Reply:
That makes sense. I will make the change.